### PR TITLE
📖 Update docs for Signed-Releases check

### PR DESCRIPTION
**Pull request by raghavkaul**
#### What kind of change does this PR introduce?
Update docs to clarify that the Signed-Releases check ignores source code-only releases.

Addresses #3439.

### CODE REVIEW

**Comment by diogoteles08** [comment]
Thanks for the PR! It indeed makes things clearer.

I'd just recommend a rephrase of the initial text, because in that version the verb "looks" can provoke some readers (I felt it hehe) to think it refers to "releases". Also I was not familiar with the "bundle", so I'd also change to something simpler.

My suggestion: `
This check looks for the most recent release associated to an artifact. It ignores the source code-only releases that are created automatically by GitHub.
`

Initial text: `
This check ignores source code-only releases that are bundled by GitHub and looks for the most recent release with a associated artifact.
`

File: docs/checks/internal/checks.yaml [hunk 1]
```diff
@@ -630,6 +630,8 @@ checks:
       If a signature is found in the assets for each release, a score of 8 is given.
       If a [SLSA provenance file](https://slsa.dev/spec/v0.1/index) is found in the assets for each release (*.intoto.jsonl), the maximum score of 10 is given.
 
+      This check looks for the most recent releases associated with an artifact. It ignores the source code-only releases that are created automatically by GitHub.
```
Comment (spencerschrock) [inline docs/checks/internal/checks.yaml:633]: there may be some confusion about what the most recent means. I think it's 30 for github by defualt?

**Review by spencerschrock — APPROVED** [review]

File: docs/checks/internal/checks.yaml (hunk 1 above)
Comment (diogoteles08) [reply docs/checks/internal/checks.yaml:633]: Good point, it's better to specify the number of days

Comment (spencerschrock) [reply docs/checks/internal/checks.yaml:633]: I don't think it's related to the number of days currently.

**Pull request merged**